tt_metal/llrt/tt_memory.cpp

```diff
@@ -4,14 +4,15 @@
 
 #include "tt_memory.h"
 
-#include <cassert>
 #include <cstdio>
-#include <fstream>
+#include <cstdint>
+#include <cstring>
+#include <functional>
 #include <limits>
-#include <stdexcept>
+#include <string>
+#include <span> // FIXME: No C++20 desired in tt_metal
```

Comment:
outside the scope of this PR but can this be replaced with `tt_metal/tt_stl/span.hpp`